plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/IcebergMetadata.java

```diff
@@ -1530,6 +1530,12 @@ public void dropColumn(ConnectorSession session, ConnectorTableHandle tableHandl
         IcebergColumnHandle handle = (IcebergColumnHandle) column;
         Table icebergTable = catalog.loadTable(session, ((IcebergTableHandle) tableHandle).getSchemaTableName());
         try {
+            icebergTable.spec().fields().stream()
+                    .filter(field -> field.name().equals(handle.getName()))
+                    .findAny()
+                    .ifPresent(partitionField -> {
+                        throw new TrinoException(NOT_SUPPORTED, "Can not drop partition field: " + partitionField);
```

Comment:
`PartitionField` object in the error message is an internal representation. Just mentioning the column name (or without the column name) is better in my opinion.